File: src/main/java/ar/edu/unq/epers/unidad5/model/Item.java
package ar.edu.unq.epers.unidad5.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.List;

@ToString
@Setter
@Getter
@EqualsAndHashCode
@NoArgsConstructor(access = AccessLevel.PROTECTED)

public class Item {
    private String nombre;
    private int peso;
    private List<Dungeon> obtenibleEn = new ArrayList<>();

    public Item(String nombre, int peso, List<Dungeon> obtenibleEn) {
        this.nombre = nombre;
        this.peso = peso;
        this.obtenibleEn = obtenibleEn;
    }
}

File: src/main/kotlin/ar/edu/unq/epers/unidad5/model/Item.kt
package ar.edu.unq.epers.unidad5.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
class Item {
    @Id
    var nombre: String = ""
    var peso: Int = 0
    var obtenibleEn: MutableList<Dungeon> = mutableListOf()

    constructor(nombre: String, peso: Int, obtenibleEn: MutableList<Dungeon>) {
        this.nombre = nombre
        this.peso = peso
        this.obtenibleEn = obtenibleEn
    }

}